i6_models/parts/conformer/convolution.py

```diff
@@ -85,3 +90,78 @@ def forward(self, tensor: torch.Tensor) -> torch.Tensor:
         tensor = self.pointwise_conv2(tensor)
 
         return self.dropout(tensor)
+
+
+@dataclass
+class ConformerConvolutionV2Config(ConformerConvolutionV1Config):
+    """
+    New attribute:
+        dropout_broadcast_axes: string of axes to which dropout is broadcast, e.g. "T" for broadcasting to the time axis
+                                setting to None to disable broadcasting
+    Allows even kernel size
+    """
+
+    dropout_broadcast_axes: Optional[str] = None
+
+    def check_valid(self):
+        assert self.kernel_size % 2 == 1, "ConformerConvolutionV1 only supports odd kernel sizes"
+
+        assert self.dropout_broadcast_axes is None or self.dropout_broadcast_axes in [
+            "B",
+            "T",
+            "BT",
+        ], "invalid value, supported are None, 'B', 'T' and 'BT'"
+
+
+class ConformerConvolutionV2(ConformerConvolutionV1):
+    """
+    Augments ConformerMHSAV1 with dropout broadcasting
+    """
+
+    def __init__(self, model_cfg: ConformerConvolutionV2Config):
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__(model_cfg)
+
+        self.dropout_broadcast_axes = model_cfg.dropout_broadcast_axes
+        self.dropout = (
+            nn.Dropout1d(model_cfg.dropout) if model_cfg.dropout_broadcast_axes else nn.Dropout(model_cfg.dropout)
+        )
+
+    def forward(self, tensor: torch.Tensor) -> torch.Tensor:
+        """
+        :param tensor: input tensor of shape [B,T,F]
+        :return: torch.Tensor of shape [B,T,F]
+        """
+        tensor = self.layer_norm(tensor)
+        tensor = self.pointwise_conv1(tensor)  # [B,T,2F]
+        tensor = nn.functional.glu(tensor, dim=-1)  # [B,T,F]
+
+        # conv layers expect shape [B,F,T] so we have to transpose here
+        tensor = tensor.transpose(1, 2)  # [B,F,T]
+        tensor = self.depthwise_conv(tensor)
+
+        tensor = self.norm(tensor)
+        tensor = tensor.transpose(1, 2)  # transpose back to [B,T,F]
+
+        tensor = self.activation(tensor)
+        tensor = self.pointwise_conv2(tensor)
+
+        if self.dropout_broadcast_axes is None:
+            tensor = self.dropout(tensor)
+        elif self.dropout_broadcast_axes == "T":
+            tensor = self.dropout(tensor.transpose(1, 2)).transpose(1, 2)
+        elif self.dropout_broadcast_axes == "B":
+            tensor = self.dropout(tensor.permute(1, 2, 0)).permute(2, 0, 1)
+        elif self.dropout_broadcast_axes == "BT":
+            batch_dim_size = tensor.shape[0]
+            feature_dim_size = tensor.shape[-1]
+
+            tensor = (
+                self.dropout(tensor.reshape(-1, feature_dim_size).transpose(0, 1))
+                .transpose(0, 1)
+                .reshape(batch_dim_size, -1, feature_dim_size)
+            )
```

Comment:
I would suggest to have our own `Dropout` module which support this `dropout_broadcast_axes` argument, and then does this logic, but better commented. The code here is really hard to follow, esp also because there are no comments on what shape you get. Then the code here simply becomes:
```suggestion
        tensor = self.dropout(tensor)
```

Reply:
do you any objection to setting the default behavior to no broadcasting?